django/contrib/auth/__init__.py

```diff
@@ -155,8 +157,15 @@ def login(request, user, backend=None):
     the anonymous session is retained when the user logs in.
     """
     session_auth_hash = ""
+    # RemovedInDjango61Warning.
     if user is None:
         user = request.user
+        warnings.warn(
+            "Fallback to request.user when user is None will be removed.",
+            RemovedInDjango61Warning,
+            stacklevel=2,
+        )
+
```

Comment:
```suggestion
    # RemovedInDjango61Warning: When the deprecation ends, replace with:
    # session_auth_hash = user.get_session_auth_hash()
```
I believe this means we will also lose the guard on whether `get_session_auth_hash` exists as we are making user a required argument and it should be a user